### PR TITLE
Update DB_query_builder.php - phpdoc (some incorrect return type)

### PR DESCRIPTION
**Pull request by damiengrandi**
Edit of some return type in phpdoc. This can also help to fix some autocompletion troubles in some IDE.

### CODE REVIEW

**Comment by mwhitneysdsu** [comment]
Most of the return types you've marked as `CI_DB_query_builder` would probably return `CI_DB` at runtime. The `delete()` method would return `string|bool|void`. There may be some additional variations based on the driver in use, as well.

File: system/database/DB_query_builder.php [hunk 1]
```diff
@@ -1456,7 +1456,7 @@ public function get_where($table = '', $where = NULL, $limit = NULL, $offset = N
 	 * @param	string	$table	Table to insert into
 	 * @param	array	$set 	An associative array of insert values
 	 * @param	bool	$escape	Whether to escape values and identifiers
-	 * @return	int	Number of rows inserted or FALSE on failure
+	 * @return	int|bool	 Number of rows inserted or FALSE on failure
```
Comment (narfbg) [inline system/database/DB_query_builder.php]: We don't really use this format, and the rare case of boolean FALSE returns has been discussed before - it's practically irrelevant... please revert this.

File: system/database/DB_query_builder.php [hunk 2]
```diff
@@ -1863,7 +1863,7 @@ protected function _validate_update($table)
 	 * @param	string	the table to retrieve the results from
 	 * @param	array	an associative array of update values
 	 * @param	string	the where key
-	 * @return	int	number of rows affected or FALSE on failure
+	 * @return	int|bool	 Number of rows affected or FALSE on failure
```
Comment (narfbg) [inline system/database/DB_query_builder.php]: Same as for `insert_batch()`.

File: system/database/DB_query_builder.php [hunk 3]
```diff
@@ -2114,7 +2114,7 @@ public function get_compiled_delete($table = '', $reset = TRUE)
 	 * @param	mixed	the where clause
 	 * @param	mixed	the limit clause
 	 * @param	bool
-	 * @return	mixed
+	 * @return	CI_DB_query_builder|bool    CI_DB_query_builder instance or FALSE on failure
```
Comment (narfbg) [inline system/database/DB_query_builder.php]: It can also return void ... mixed is what should stay here.

**Comment by narfbg** [comment]
`CI_DB` would still be a valid `CI_DB_query_builder` instance in all of these cases - don't worry about that.

**Comment by damiengrandi** [comment]
Ok, I updated

**Pull request merged**